frontend/components/user-settings.tsx

```diff
@@ -0,0 +1,388 @@
+
+  const handleProfilePictureChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0]
+    if (file) {
+      const reader = new FileReader()
+      reader.onloadend = () => {
+        setProfilePicture(reader.result as string)
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const handleDeleteProfilePicture = () => {
+    setProfilePicture('/placeholder.svg?height=100&width=100')
+  }
+
+  const handleSaveChanges = async () => {
+    if (user) {
+      const token = localStorage.getItem('jwtToken')
+      if (!token) {
+        console.error('No authentication token found')
+        return
+      }
+
+      try {
+        const response = await fetch(`http://localhost:3001/users/${userId}`, {
+          method: 'PATCH',
+          headers: {
+            'Authorization': `Bearer ${token}`,
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(user),
+        })
+        if (!response.ok) throw new Error('Failed to save changes')
+
+        console.log('Changes saved successfully!')
+        mutate()
+      } catch (error) {
+        console.error('Error saving changes:', error)
+        // Handle error (e.g., show an error message to the user)
+      }
+    }
+  }
+
+  if (isLoading) {
+    return <div>Loading user data...</div>
+  }
+
+  if (error) {
+    return <div>Error: Failed to load user data</div>
+  }
+
+  if (!user) {
+    return <div>No user data available.</div>
+  }
+
+  const handlePasswordInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { id, value } = e.target
+    switch (id) {
+      case 'currentPassword':
+        setCurrentPassword(value)
+        break
+      case 'newPassword':
+        setNewPassword(value)
+        break
+      case 'confirmPassword':
+        setConfirmPassword(value)
+        break
+    }
+  }
+
+  const verifyCurrentPassword = async (email: string, password: string) => {
+    try {
+      const response = await fetch('http://localhost:3001/auth/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, password }),
+      });
+
+      if (response.ok) {
+        return true; // Password is correct
+      } else {
+        return false; // Password is incorrect
+      }
+    } catch (error) {
+      console.error('Error verifying password:', error);
+      return false;
+    }
+  };
+
+  /**
+ * Handles the change password process.
+ *
+ * Validates the input fields, verifies the current password, and updates the password if all checks pass.
+ *
+ * @returns {Promise<void>} - Returns a promise that resolves when the password change process is complete.
+ *
+ * @example
+ * await handleChangePassword();
+ */
+
+  const handleChangePassword = async () => {
+    // Validate inputs
+    if (!currentPassword || !newPassword || !confirmPassword) {
+      alert('Please fill in all password fields');
```

Comment:
We can perhaps use `toast` from shacdn UI instead of alert.

Reply:
Same goes for all other `alerts` in the file

Reply:
agreed